File: mobile/lib/dtos/requests/expense_creation_request.dart
class ExpenseCreationRequest {
  final int paymentMethodId;
  final int currencyId;
  final double amount;
  final String extraInfo;
  final DateTime expenseDate;
  final int categoryId;
  final int? cardId;

  ExpenseCreationRequest({
    required this.paymentMethodId,
    required this.currencyId,
    required this.amount,
    required this.extraInfo,
    required this.expenseDate,
    required this.categoryId,
    this.cardId,
  });

  Map<String, dynamic> toJson() {
    return {
      'paymentMethodId': paymentMethodId,
      'currencyId': currencyId,
      'amount': amount,
      'extraInfo': extraInfo,
      'date': expenseDate.toIso8601String(),
      'categoryId': categoryId,
      if (cardId != null) 'cardId': cardId,
    };
  }

}

File: web/Finance/src/main/java/com/github/Finance/dtos/requests/ExpenseCreationRequest.java
package com.github.Finance.dtos.requests;

import java.time.LocalDate;

/**
 * This is a complex DTO for creating a simple expenses. Many business rules must apply to it, but
 * it is the service who must check them
 */
public record ExpenseCreationRequest(
    Long paymentMethodId,
    Long currencyId,
    Double amount,
    String extraInfo,
    LocalDate date,
    Long categoryId,
    Boolean isSubscription,
    Integer installments,
    Integer cardId
){
}
